h2/src/main/org/h2/command/Parser.java

```diff
@@ -4138,6 +4154,8 @@ private Column parseColumnWithType(String columnName) {
             precision = templateColumn.getPrecision();
             displaySize = templateColumn.getDisplaySize();
             scale = templateColumn.getScale();
+        } else if (customDataType != null) {
+            dataType = DataType.getTypeByName("JAVA_OBJECT");
```

Comment:
Probably we should let implementations declare what type they boil down to, what do you think? Here `JAVA_OBJECT` is present simply as long as it is the most generic thing possible.